server/issue.go

```diff
@@ -1127,21 +1130,18 @@ func getUpdateGhostTaskList(database *api.Database, vcsPushEvent *vcs.PushEvent,
 			Payload:           string(bytes),
 		})
 	}
-	// ID is actually the taskCreateList array index here because we don't know task ID before inserting.
-	// So the below list means that taskCreateList[0] blocks taskCreateList[1], taskCreateList[1] blocks taskCreateList[2], which makes sense because the "sync" task blocks the "cutover" task, and the "drop original table" task can only be done after "cutover".
-	taskDAGCreateList := []api.TaskDAGCreate{
+	// The below list means that taskCreateList[0] blocks taskCreateList[1], taskCreateList[1] blocks taskCreateList[2], which makes sense because the "sync" task blocks the "cutover" task, and the "drop original table" task can only be done after "cutover".
+	taskIndexDAGList := []api.TaskIndexDAG{
 		{
-			FromTaskID: 0,
-			ToTaskID:   1,
-			Payload:    "{}",
+			FromIndex: 0,
```

Comment:
Shall we pass in as starting index if we have multiple databases in a stage?

Reply:
Btw, I'm thinking more of a tenant-mode case.

Reply:
It is for sure that tenant mode will require some modifications to this method, but it's another story. This is enough for non-tenant mode for now. I think the modifications you mentioned shall be done in another PR.

Reply:
SG.